x-pack/plugins/lens/public/state_management/init_middleware/load_initial.ts

```diff
@@ -28,13 +29,37 @@ export const getPersisted = async ({
 }: {
   initialInput: LensEmbeddableInput;
   lensServices: LensAppServices;
-}): Promise<{ doc: Document } | undefined> => {
-  const { notifications, attributeService } = lensServices;
+}): Promise<
+  { doc: Document; sharingSavedObjectProps: Omit<SharingSavedObjectProps, 'errorJSON'> } | undefined
+> => {
+  const { notifications, spaces, attributeService } = lensServices;
   let doc: Document;
 
   try {
-    const attributes = await attributeService.unwrapAttributes(initialInput);
-
+    const result = await attributeService.unwrapAttributes(initialInput);
+    if (!result) {
+      return {
+        doc: ({
+          ...initialInput,
+          type: LENS_EMBEDDABLE_TYPE,
+        } as unknown) as Document,
+        sharingSavedObjectProps: {
+          outcome: 'exactMatch',
+        },
+      };
+    }
+    const { sharingSavedObjectProps, ...attributes } = result;
+    if (spaces && sharingSavedObjectProps?.outcome === 'aliasMatch') {
+      // We found this object by a legacy URL alias from its old ID; redirect the user to the page with its new ID, preserving any URL hash
+      const newObjectId = sharingSavedObjectProps?.aliasTargetId; // This is always defined if outcome === 'aliasMatch'
+      const newPath = lensServices.http.basePath.prepend(getFullPath(newObjectId));
```

Comment:
Same as my other comment above, I think you need to use `getEditPath` and try to keep the global filter intact.